pkg/libdevfile/libdevfile.go

```diff
@@ -69,6 +87,214 @@ func getDefaultCommand(devfileObj parser.DevfileObj, kind v1alpha2.CommandGroupK
 	return groupCmds[0], nil
 }
 
+// ValidateAndGetPushCommands validates the build and the run command,
+// if provided through odo push or else checks the devfile for devBuild and devRun.
+// It returns the build and run commands if its validated successfully, error otherwise.
+func ValidateAndGetPushCommands(
+	data data.DevfileData,
+	devfileBuildCmd,
+	devfileRunCmd string,
+) (commandMap map[v1alpha2.CommandGroupKind]v1alpha2.Command, err error) {
+	var emptyCommand v1alpha2.Command
+	commandMap = make(map[v1alpha2.CommandGroupKind]v1alpha2.Command)
+
+	isBuildCommandValid, isRunCommandValid := false, false
+
+	buildCommand, buildCmdErr := GetBuildCommand(data, devfileBuildCmd)
+
+	isBuildCmdEmpty := reflect.DeepEqual(emptyCommand, buildCommand)
+	if isBuildCmdEmpty && buildCmdErr == nil {
+		// If there was no build command specified through odo push and no default build command in the devfile, default validate to true since the build command is optional
```

Comment:
```suggestion
		// If there was no build command specified through odo dev and no default build command in the devfile, default validate to true since the build command is optional
```